src/features/posts/Post/Post.tsx

```diff
@@ -1,27 +1,31 @@
-import React, { useMemo } from 'react';
+import React from 'react';
+import { useParams } from 'react-router-dom';
+import { useSelector } from 'react-redux';
 
-import { PostBody } from '../PostBody/PostBody';
+import { PostBody } from 'features/posts/PostBody/PostBody';
+
+import { RootState } from 'app/store';
 
 export interface PostProps {}
 
 export const Post: React.FC<PostProps> = () => {
   // TODO: Get the ID from the router params (e.g. /posts/12 => 12) and get the
   // post from the Redux store that matches the ID to replace this test data.
-  const post = useMemo(
-    () => ({
-      userId: 1,
-      id: 2,
-      title: 'qui est esse',
-      body:
-        'est rerum tempore vitae\nsequi sint nihil reprehenderit dolor beatae ea dolores neque\nfugiat blanditiis voluptate porro vel nihil molestiae ut reiciendis\nqui aperiam non debitis possimus qui neque nisi nulla',
-    }),
-    []
+  const params: {
+    id: string | undefined;
+  } = useParams();
+  const { id } = params;
+  console.log('params', useParams());
+  const post = useSelector((state: RootState) =>
+    state.posts.posts.find((post: any) => post.id === Number(id))
```

Comment:
you already typed `posts` in `postsSlice.ts`, so you don't need to type this post anymore and what's more important -definitely you don't want to type it as `any`